### ISSUE TITLE
False negative ForbiddenComment with common TODO markers

### ISSUE THREAD

**Issue by TWiStErRob** [detekt/detekt#6116]
The power of `ForbiddenComment` is increased with #5981 and therefore we can decrease the false negatives, while not introducing new false positives.

## Expected Behavior
Flag all of:
```kotlin
// TODO: foo
// FIXME: foo
// STOPSHIP: foo
// TODO foo
// FIXME foo
// STOPSHIP foo
```
and by extension
```kotlin
// TODO : foo
// FIXME : foo
// STOPSHIP : foo
```
([French grammar mandates a space](https://en.wikipedia.org/wiki/Colon_(punctuation)#Spacing_and_parentheses) before the `:` so this is common by French speakers; I've seen it from French and Moroccan coders.)

Potentially these too:
```kotlin
// todo foo
// fixme foo
// stopship foo
```

All of the above can be achieved by using a regex: `(?i)\bTODO\b` similar to IDEA config. The `\b` is extra to make sure `TODOs`, `mastodon`, `affixment` words in comments are not flagged.

---

Additionally I suggest to merge the three comments into one (again this comes as a bonus of #5981 giving more power):
```yaml
ForbiddenComment:
  comments:
    - value: '(?i)\b(TODO|FIXME|STOPSHIP)\b'
      reason: 'Todo marker found, fix the code and remove comment.'
```
so that it's easier to configure the rule (no need to copy/paste and maintain 3 rules when adding new).

## Observed Behavior
These are flagged:
```kotlin
// TODO: foo
// FIXME: foo
// STOPSHIP: foo
```
nothing else.

## Steps to Reproduce
Use comments in "Expected Behavior" with default Detekt config.

https://github.com/detekt/detekt/blob/67d2c7237908c7fb069df0e914c26010b0529302/detekt-core/src/main/resources/default-detekt-config.yml#L575-L578

## Context
While working on https://github.com/detekt/detekt/pull/5981#discussion_r1198817598 I realized `// FIXME foo` would just be accepted compliant by `ForbiddenComment` rule. From experience with multiple codebases most of the TODOs developers write don't contain the colon (`:`) character.

Note: the default todo markers in IntelliJ IDEA also don't use `:`

https://www.jetbrains.com/help/idea/using-todo.html#add_pattern_filter_todo

## Your Environment

* Version of detekt used: 1.23.0
* Version of Gradle used (if applicable): N/A
* Gradle scan link (add `--scan` option when running the gradle task): N/A
* Operating System and version: N/A
* Link to your project (if it's a public repository): any

**Comment by schalkms** [comment]
duplicate #6116 ?

**Comment by TWiStErRob** [comment]
No, the fix of #3508 (which is #5981) enables this improvement. Just look at the "Expected/Observed" examples, the rest is suggestions.

**Comment by BraisGabin** [comment]
I like the idea but that regex is scary. I don't think that such a difficult regex should be part of the default configuration of detekt. Nearly no one will be brave to edit it.

I think that #6075 could help to fix this.

**Comment by BraisGabin** [comment]
What do we think about this? A change in our default configuration is not a breaking change so we can make the change propesed by @TWiStErRob and if in the end we implement #6075 we can change it again at any moment. I would approve a PR doing this change.

**Comment by TWiStErRob** [comment]
> Nearly no one will be brave to edit it.

I'm happy to write some docs on the rule on what the regex parts mean (in addition to a link to https://regex101.com/r/IEOm2Q/1). And I think even the some programmer without any knowledge of RegEx would be able to figure out how to add/remove their own custom todo markers.

**Comment by BraisGabin** [comment]
Go for it :)
